app.py

```diff
@@ -197,6 +198,9 @@ def initialize_session():
     utils.session_id()
     login_user_from_token_cookie()
 
+    # Creation of test_group field if it doesn't exist in session or db user
+    alternative_error_messages.update_users_test_group(g.db)
```

Comment:
This is called before each request. Can't we just call this function once in the `/login` route?

Reply:
We want to maximize the percentage of users which are used for the test group selection. So if we place it in the /login route, all the users with active session don't be available.

Reply:
Doing that only on `/login` will not cover all the sessions handled by `login_user_from_token_cookie()`. I think this really does the minimum required to reliably get the test group flag.